jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/KryoBooster.java

```diff
@@ -0,0 +1,50 @@
+package ml.dmlc.xgboost4j.java;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.KryoSerializable;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.util.Map;
+
+@SuppressWarnings("unused")
+public class KryoBooster extends Booster implements KryoSerializable {
+
+  private static final Log logger = LogFactory.getLog(KryoBooster.class);
+
+  KryoBooster(Map<String, Object> params, DMatrix[] cacheMats) throws XGBoostError {
+    super(params, cacheMats, true);
+  }
+
+  @Override
+  public void write(Kryo kryo, Output output) {
+    try {
+      byte[] serObj = this.toByteArray();
+      int serObjSize = serObj.length;
+      System.out.println("==== serialized obj size " + serObjSize);
+      output.writeInt(serObjSize);
+      output.write(serObj);
+    } catch (XGBoostError ex) {
+      ex.printStackTrace();
```

Comment:
Are you sure you would like to swallow the exception?

Reply:
This is a copy&paste of what was originally in the methods - I don't know why they did that way - seems wrong. Thanks for noticing - we should probably throw an exception.